### PR TITLE
pool fee bridge fix

### PR DESCRIPTION
**Pull request by zaryab2000**

### CODE REVIEW

File: contracts/PushComm/PushCommV3.sol [hunk 1]
```diff
@@ -689,21 +689,33 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
      */
     // Should be only admin
     // Should only bridge NTT TOKENS FROM COMM TO CORE on ethereum
-    function transferFeePoolToCore(uint256 _amount) external payable onlyPushChannelAdmin {
+    function transferFeePoolToCore(CrossChainRequestTypes.CrossChainFunction functionType, uint256 _amount, uint256 gasLimit) external payable onlyPushChannelAdmin {
```
Comment (0xNilesh) [inline contracts/PushComm/PushCommV3.sol:692]: no need of taking functionType in the param, AdminRequest_AddPoolFee can be used directly here in this func

File: contracts/PushComm/PushCommV3.sol [hunk 2]
```diff
@@ -689,21 +689,33 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
...
 
         PUSH_NTT.approve(address(NTT_MANAGER), _amount);
-        NTT_MANAGER.transfer{ value: tokenBridgeCost }(_amount, WORMHOLE_RECIPIENT_CHAIN, recipient);
+        NTT_MANAGER.transfer{ value: tokenBridgeCost }(_amount, WORMHOLE_RECIPIENT_CHAIN, BaseHelper.addressToBytes32(EPNSCoreAddress), BaseHelper.addressToBytes32(msg.sender), false, new bytes(1));
+        
+        bytes memory requestPayload = abi.encode(functionType, bytes(""), _amount, msg.sender);
+
+         // Relay the RequestData Payload
+        WORMHOLE_RELAYER.sendPayloadToEvm{ value: messageBridgeCost }(
+            WORMHOLE_RECIPIENT_CHAIN,
```
Comment (0xNilesh) [inline contracts/PushComm/PushCommV3.sol:712]: GAS#2: WORMHOLE_RECIPIENT_CHAIN can be stored in memory and then used to minimize gas usage

File: contracts/PushComm/PushCommV3.sol [hunk 3]
```diff
@@ -689,21 +689,33 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
...
-
+        uint256 messageBridgeCost = quoteMsgRelayCost(WORMHOLE_RECIPIENT_CHAIN, gasLimit);
         uint256 tokenBridgeCost = quoteTokenBridgingCost();
-        if (msg.value < tokenBridgeCost) {
+
+        if (msg.value < (messageBridgeCost + tokenBridgeCost)) {
             revert Errors.InsufficientFunds();
         }
 
         PROTOCOL_POOL_FEE = PROTOCOL_POOL_FEE - _amount;
```
Comment (0xNilesh) [inline contracts/PushComm/PushCommV3.sol:703]: GAS#1: PROTOCOL_POOL_FEE can also be stored in memory and then updated in the last

File: contracts/PushComm/PushCommV3.sol [hunk 4]
```diff
@@ -689,21 +689,33 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
...
 
         PUSH_NTT.approve(address(NTT_MANAGER), _amount);
-        NTT_MANAGER.transfer{ value: tokenBridgeCost }(_amount, WORMHOLE_RECIPIENT_CHAIN, recipient);
+        NTT_MANAGER.transfer{ value: tokenBridgeCost }(_amount, WORMHOLE_RECIPIENT_CHAIN, BaseHelper.addressToBytes32(EPNSCoreAddress), BaseHelper.addressToBytes32(msg.sender), false, new bytes(1));
```
Comment (0xNilesh) [inline contracts/PushComm/PushCommV3.sol:706]: GAS#3: EPNSCoreAddress can be stored in memory and then use

File: contracts/PushComm/PushCommV3.sol [hunk 5]
```diff
@@ -689,21 +689,33 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
...
         uint256 tokenBridgeCost = quoteTokenBridgingCost();
-        if (msg.value < tokenBridgeCost) {
+
+        if (msg.value < (messageBridgeCost + tokenBridgeCost)) {
             revert Errors.InsufficientFunds();
         }
 
         PROTOCOL_POOL_FEE = PROTOCOL_POOL_FEE - _amount;
 
         PUSH_NTT.approve(address(NTT_MANAGER), _amount);
```
Comment (0xNilesh) [inline contracts/PushComm/PushCommV3.sol:705]: Gas#4: NTT_MANAGER interface could be stored in a memory variable and then use

File: contracts/PushComm/PushCommV3.sol (hunk 1 above)
Comment (0xNilesh) [inline contracts/PushComm/PushCommV3.sol:692]: either have all parameters with a '_' prefix or none for consistency

**Review by 0xNilesh — COMMENTED** [review]
Functionality wise LGTM

File: contracts/PushComm/PushCommV3.sol (hunk 2 above)
Comment (zaryab2000) [reply contracts/PushComm/PushCommV3.sol:712]: Agreed. Also the same was also true for the previously included function `_createCrossChainRequest`. Therefore I have fixed that too now. @0xNilesh

Comment (0xNilesh) [reply contracts/PushComm/PushCommV3.sol:712]: ah okay, btw these fixes for createCrossChainRequest fn are added in this Pr: https://github.com/push-protocol/push-smart-contracts/pull/394

File: contracts/PushComm/PushCommV3.sol (hunk 4 above)
Comment (zaryab2000) [reply contracts/PushComm/PushCommV3.sol:706]: done

File: contracts/PushComm/PushCommV3.sol (hunk 1 above)
Comment (zaryab2000) [reply contracts/PushComm/PushCommV3.sol:692]: done

File: contracts/PushComm/PushCommV3.sol (hunk 5 above)
Comment (zaryab2000) [reply contracts/PushComm/PushCommV3.sol:705]: why? NTT_MANAGER state is only used once. @nil

Comment (0xNilesh) [reply contracts/PushComm/PushCommV3.sol]: 1. https://github.com/push-protocol/push-smart-contracts/blob/796b637986b8a3aa1f5841395dec463c520186dd/contracts/PushComm/PushCommV3.sol#L659
2. https://github.com/push-protocol/push-smart-contracts/blob/796b637986b8a3aa1f5841395dec463c520186dd/contracts/PushComm/PushCommV3.sol#L660

Comment (zaryab2000) [reply contracts/PushComm/PushCommV3.sol]: the one on line 4 is a function call using NTT_MANAGER. will it help using local variable in this case...if we have the gas data on this we can include this change else skip. @0xNilesh

Comment (0xNilesh) [reply contracts/PushComm/PushCommV3.sol]: I tried it a few days ago in remix, and it definitely helps
I can get you results if you need

**Pull request merged**